src/cli/java/org/commcare/util/cli/ApplicationHost.java

```diff
@@ -323,14 +341,17 @@ private void printErrorAndContinue(String error, Exception e) {
 
     private Screen getNextScreen() {
         String next = mSession.getNeededData(mSession.getEvaluationContext());
-
         if (next == null) {
             //XFORM TIME!
             return null;
         } else if (next.equals(SessionFrame.STATE_COMMAND_ID)) {
             return new MenuScreen();
         } else if (next.equals(SessionFrame.STATE_DATUM_VAL)) {
-            return new EntityScreen();
+            return new EntityScreen(true);
+        } else if (next.equals(SessionFrame.STATE_QUERY_REQUEST)) {

```

Comment:
Why is it that these are just getting added now when this functionality has existed for so long?

Reply:
Web Apps doesn't use `ApplicationHost` as its runner, `MenuSession` has its own control logic